src/full/Agda/TypeChecking/Rules/Application.hs

```diff
@@ -1044,23 +1044,23 @@ inferOrCheckProjApp e o ds args mt = do
         [ "  principal arg " <+> prettyTCM arg
         , "  has type "      <+> prettyTCM ta
         ]
-      inferOrCheckProjAppToKnownPrincipalArg e o ds args mt k v0 ta
+      inferOrCheckProjAppToKnownPrincipalArg e o ds args mt k v0 ta Nothing
 
 -- | Same arguments 'inferOrCheckProjApp' above but also gets the position,
 --   value and type of the principal argument.
 inferOrCheckProjAppToKnownPrincipalArg ::
   A.Expr -> ProjOrigin -> List1 QName -> A.Args -> Maybe (Comparison, Type) ->
-  Int -> Term -> Type -> TCM (Term, Type, CheckedTarget)
-inferOrCheckProjAppToKnownPrincipalArg e o ds args mt k v0 ta = do
+  Int -> Term -> Type -> Maybe (Args, Type) -> TCM (Term, Type, CheckedTarget)
```

Comment:
Please document the purpose of the new argument.
(Maybe a good opportunity to document the other arguments as well.)

Reply:
Yup, good point. I documented all arguments.

Reply:
Great!